### ISSUE TITLE
Performance degradation with JSON introduce

### ISSUE THREAD

**Issue by kshcherbatov** [tarantool/tarantool#4102]
Tarantool version: 2.1

With commit 4273ec52e122d6d37c8deedf1bc10732a7e40c0e Tarantool performance on sysbech test for SQL has decreased. This is due to the increasing complexity of individual code fragments. We have to understand why **SQL** performance has decreased, what is a major contribution to this problem, whether can we do something with it.

![photo_2019-04-04_17-35-09](https://user-images.githubusercontent.com/9724311/55622561-3a41ef80-57a9-11e9-8894-9453b2256139.jpg)

**Comment by kshcherbatov** [comment]
I've profiled tarantool on this workload before JSON path 4273ec5 and after an reduced that the main impact of this issue is belongs to tuple_format_cmp;
This routine is a part of "format reuse" #3924 feature.

Due to the fact that this is a SQL test, ephemeral spaces are often created in the process of work, for which a format is required. After creating the format, an attempt is made to reuse the existing one, during which tuple_format_cmp is called. Accordingly, the creation of ephemeral spaces has become much slower, although a JSON tree is never required to serve them.

To test this hypothesis, I made a small patch that eliminates these complex checks(since SHL tests never create JSON indexes, no problems will arise). Performance was almost completely restored.

![photo_2019-04-05_15-19-55](https://user-images.githubusercontent.com/9724311/55627241-58621c80-57b6-11e9-82e3-5054674022d4.jpg)

![photo_2019-04-05_15-19-59](https://user-images.githubusercontent.com/9724311/55627248-5b5d0d00-57b6-11e9-90fc-386dc5cd563b.jpg)

![photo_2019-04-05_15-20-06](https://user-images.githubusercontent.com/9724311/55627253-5d26d080-57b6-11e9-9269-3ecf795bf1ae.jpg)

![photo_2019-04-05_15-20-47](https://user-images.githubusercontent.com/9724311/55627267-6adc5600-57b6-11e9-8dcc-9ded43e91ef1.jpg)
